Compiler/FrontEnd/InstUtil.mo

```diff
@@ -6063,70 +6049,72 @@ algorithm
   end match;
 end extractCurrentName;
 
-public function splitConnectEquationsExpandable
+public function reorderConnectEquationsExpandable
 "@author: adrpo
   Reorder the connect equations to have non-expandable connect first:
     connect(non_expandable, non_expandable);
     connect(non_expandable, expandable);
     connect(expandable, non_expandable);
     connect(expandable, expandable);"
-  input FCore.Cache inCache;
-  input FCore.Graph inEnv;
-  input InnerOuter.InstHierarchy inIH;
-  input Prefix.Prefix inPre;
+  input output FCore.Cache cache;
+  input FCore.Graph env;
   input list<SCode.Equation> inEquations;
-  input Boolean impl;
-  input list<SCode.Equation> inAccumulatorNonExpandable;
-  input list<SCode.Equation> inAccumulatorExpandable;
-  output FCore.Cache outCache;
   output list<SCode.Equation> outEquations;
-  output list<SCode.Equation> outExpandableEquations;
+protected
+  DoubleEndedList<SCode.Equation> delst;
+  list<SCode.Equation> expandableEqs;
+  Absyn.ComponentRef crefLeft, crefRight;
+  DAE.Type ty1,ty2;
 algorithm
-  (outCache,outEquations,outExpandableEquations) := matchcontinue(inCache, inEnv, inIH, inPre, inEquations, impl, inAccumulatorNonExpandable, inAccumulatorExpandable)
-    local
-      list<SCode.Equation>  rest, eEq, nEq;
-      SCode.Equation eq;
-      Absyn.ComponentRef crefLeft, crefRight;
-      FCore.Cache cache;
-      FCore.Graph env;
-      SourceInfo info;
-      DAE.Type ty1,ty2;
-      DAE.ComponentRef c1_1,c2_1;
-
-    // if we have no expandable connectors, return the same
-    case (cache, _, _, _, _::_, _, _, _)
-      equation
-        false = System.getHasExpandableConnectors();
-      then
-        (cache, inEquations, {});
-
-    // handle empty case
-    case (cache, _, _, _, {}, _, eEq, nEq) then (cache, listReverse(eEq), listReverse(nEq));
-
-    // connect, both expandable
-    case (cache, env, _, _, (eq as SCode.EQUATION(SCode.EQ_CONNECT(crefLeft, crefRight, _, info)))::rest, _, eEq, nEq)
-      equation
-        (_, ty1, _) = Lookup.lookupConnectorVar(env, ComponentReference.toExpCref(crefLeft));
-        (_, ty2, _) = Lookup.lookupConnectorVar(env, ComponentReference.toExpCref(crefRight));
-        //(cache,SOME((DAE.CREF(),DAE.PROP(ty1,_),_))) = Static.elabCref(cache, env, crefLeft, impl, false, inPre, info);
-        //(cache,SOME((DAE.CREF(),DAE.PROP(ty2,_),_))) = Static.elabCref(cache, env, crefRight, impl, false, inPre, info);
+  if if listEmpty(inEquations) then true else (not System.getHasExpandableConnectors()) then
+    outEquations := inEquations;
+    return;
+  end if;
+  ErrorExt.setCheckpoint("expandableConnectorsOrder");
+  delst := DoubleEndedList.fromList({});
```

Comment:
I doubt using double ended list instead of consecutive prepending an item to empty list + listReverseInPlace is any faster or less memory consuming. It's like shooting sparrows with a cannon...